components/micro-gateway-core/src/main/java/org/wso2/micro/gateway/core/jwtgenerator/AbstractMGWJWTGenerator.java

```diff
@@ -0,0 +1,366 @@
+package org.wso2.micro.gateway.core.jwtgenerator;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.FileInputStream;
+import java.nio.charset.Charset;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.MessageDigest;
+import java.security.PrivateKey;
+import java.security.Signature;
+import java.security.cert.Certificate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ *  Abstract class for generate JWT for backend claims
```

Comment:
Checkstyle failure if no `.` at the end of first sentence.
```suggestion
 *  Abstract class for generate JWT for backend claims.
```